pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,28 @@
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec   `json:"spec,omitempty"`
+	Status            TrainedModelStatus `json:"status,omitempty"`
+}
+type TrainedModelSpec struct {
+	// Required field for parent inference service
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	PredictorModel ModelSpec `json:"predictorModel"`
+}
+type ModelSpec struct {
+	// Storage URI for the model repository
+	StorageURI string `json:"storageUri"`
+	// Machine Learning <framework name>:<git tag>
+	// The values could be: "tensorflow:v2.2.0","pytorch:v1.5.1","sklearn:0.23.1","onnx:v1.7.0","xgboost:v1.1.1", "myawesomeinternalframework:1.1.0" etc.
+	Framework string `json:"framework"`
+	// Maximum memory this model will consume, this field is used to decide if a model server has enough memory to load this model.
+	Memory v1.ResourceName `json:"memory"`
```

Comment:
I removed ModelVersionPolicy section for now per @ellistarn's suggestion since we should take some time to gain more experience about model versioning. I added this section initially because I think the new MLServer we are implementing should follow Triton's model repo's structure and the model version policy, see my earlier reply here: https://github.com/kubeflow/kfserving/pull/908#discussion_r447124002.

Reply:
Interesting idea to make
>  TrainedModelSpec.InferenceService was made optional/mutable
I also proposed an idea to create a TrainedModelBinding CRD to bind TrainedModel and InferenceService. But it also means that we allow users to create TrainedModel CR without deploying. Per @yuzisun's comment earlier, if we completely decouple TrainedModel and InferenceService CRD then KFServing may become a staging area for un-deployed models. I don't want to make TrainedModelSpec.InferenceService was made optional/mutable as optional for now to avoid turning KFServing into a TrainedModel staging area. However, I think it is interesting to think about how to manage a model's life cycle.

Reply:
About how to get a precise memory usage of model, I think that's something we can add to MLServer's metadata endpoint @cliveseldon wdyt?

Reply:
Sorry @yuzliu I guess you put responses to all of my comments on this one because the others were on outdated changes. Will reply to all of them here but quote inline...
> I added this section initially because I think the new MLServer we are implementing should follow Triton's model repo's structure and the model version policy, see my earlier reply here: #908 (comment).

Yep probably best for versioning discussion to happen elsewhere, glad that this part was deferred.

> I also proposed an idea to create a TrainedModelBinding CRD to bind TrainedModel and InferenceService. But it also means that we allow users to create TrainedModel CR without deploying. Per @yuzisun's comment earlier, if we completely decouple TrainedModel and InferenceService CRD then KFServing may become a staging area for un-deployed models. I don't want to make TrainedModelSpec.InferenceService was made optional/mutable as optional for now to avoid turning KFServing into a TrainedModel staging area. However, I think it is interesting to think about how to manage a model's life cycle.

Yep I completely agree w.r.t. this separation of concerns. My idea was just a suggested compromise in response to your TrainedModelBinding proposal and motivation, but I think I'm equally happy with the tight coupling.

> About how to get a precise memory usage of model, I think that's something we can add to MLServer's metadata endpoint

That seems like a reasonable approach to me ... having a controller use that to update the `TrainedModel` Status. Though it might be considered more like operational state than model metadata (since memory use might vary between model server versions for example, and may not be known if that particular model isn't loaded at that point in time) so might be more appropriate to go in `ModelReadyResponse`?